src/core/thread/mesh_forwarder.hpp

```diff
@@ -131,23 +131,32 @@ class MeshForwarder
     void SetRxOnWhenIdle(bool aRxOnWhenIdle);
 
     /**
-     * This method sets customized Data Poll period. Only for certification test
+     * This method sets a user-specified Data Poll period.
      *
-     * @param[in]  aPeriod  The Data Poll period in milliseconds.
+     * If the value is set to zero, then the poll interval is controlled/chosen by the OpentThread stack.
+     * If the user has provided a non-zero poll period, the user value has higher priority and is used in place of
+     * poll period values chosen internally by OpenThread implementation (except during attachment).
+     *
+     * Initial/Default value for "assign poll period" is zero.
+     *
+     * @param[in]  aPeriod  The Data Poll period in milliseconds. Value zero
      *
      */
     void SetAssignPollPeriod(uint32_t aPeriod);
 
     /**
-     * This method gets the customized Data Poll period. Only for certification test
+     * This method gets the current user-specified Data Poll period.
      *
      * @returns  The Data Poll period in milliseconds.
      *
      */
     uint32_t GetAssignPollPeriod(void);
 
     /**
-     * This method sets the Data Poll period.
+     * This method sets/updates the Data Poll period.
+     *
+     * If the user has provided a non-zero assign poll period (@sa SetAssignPollPeriod), the user value has higher
+     * priority and is used in place of @p aPeriod (except during attachment).
      *
      * @param[in]  aPeriod  The Data Poll period in milliseconds.
```

Comment:
I was also considering this. 
I would suggest keeping value zero as a valid input but treat it to mean to stop periodic polling. Thoughts?